spec.md

```diff
@@ -266,14 +264,23 @@ circuit FooCircuit :
 In the above example, the public module `Foo` gets instantiated in the
 `FooCircuit` the same way a private module would.
 
-A public module can be instantiated in an external Circuit like so:
+A public module must have a well-defined interface that can be captured in its
+declaration, so all ports should be fixed. The following example shows a
+declaration of a public module.
```

Comment:
I'm not sure this sentence goes with the example below... this is basically how you are including a public module from Circuit into another circuitB

Reply:
I think this should clearly define "well-defined interface":
1. the module name will not change
2. no ports will be added or deleted
3. aggregate type ports will be lowered to ground type ports using the algorithm in section ???

Reply:
> aggregate type ports will be lowered to ground type ports using the algorithm in section ???

does such an algorithm exist yet, or does it need to be written?